collector/nodes.go

```diff
@@ -84,7 +88,9 @@ func createRoleMetric(role string) *nodeMetric {
 }
 
 var (
-	defaultNodeLabels               = []string{"cluster", "host", "name", "es_master_node", "es_data_node", "es_ingest_node", "es_client_node"}
+	defaultNodeLabels = []string{"cluster", "host", "name", "es_master_node", "es_data_node",
```

Comment:
I think the format of the node metrics would need to change to add more roles. Adding all of these labels to the nodes metrics doesn't feel like a good idea to me. This variable appears to be referenced over 100 times in this file with most references being adding all of these labels to all of the metrics. A better approach may be to expose a new metric for node roles as a gauge with a 1 or 0 based on if the node has that role. I know that's a much different change to this and more involved, but I don't think it's sustainable to keep adding labels to these metrics.

Reply:
Output updated
```
$ curl -s http://localhost:9108/metrics | grep 'elasticsearch_nodes_roles'
# HELP elasticsearch_nodes_roles Node default roles
# TYPE elasticsearch_nodes_roles gauge
elasticsearch_nodes_roles{cluster="xxxx",host="172.22.138.46",name="instance-0000000016",role="client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.22.138.46",name="instance-0000000016",role="data_content"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.22.138.46",name="instance-0000000016",role="data_hot"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.22.138.46",name="instance-0000000016",role="ingest"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.22.138.46",name="instance-0000000016",role="remote_cluster_client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.22.138.46",name="instance-0000000016",role="transform"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.22.143.78",name="instance-0000000015",role="client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.22.143.78",name="instance-0000000015",role="data_content"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.22.143.78",name="instance-0000000015",role="data_hot"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.22.143.78",name="instance-0000000015",role="ingest"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.22.143.78",name="instance-0000000015",role="remote_cluster_client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.22.143.78",name="instance-0000000015",role="transform"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.236.46",name="instance-0000000003",role="client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.236.46",name="instance-0000000003",role="data_warm"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.236.46",name="instance-0000000003",role="remote_cluster_client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.238.200",name="instance-0000000019",role="client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.238.200",name="instance-0000000019",role="master"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.238.200",name="instance-0000000019",role="remote_cluster_client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.241.205",name="instance-0000000021",role="client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.241.205",name="instance-0000000021",role="data_frozen"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.241.47",name="instance-0000000022",role="client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.241.47",name="instance-0000000022",role="data_cold"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.241.47",name="instance-0000000022",role="remote_cluster_client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.242.154",name="instance-0000000018",role="client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.242.154",name="instance-0000000018",role="master"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.242.154",name="instance-0000000018",role="remote_cluster_client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.242.162",name="instance-0000000007",role="client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.242.162",name="instance-0000000007",role="data_warm"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.242.162",name="instance-0000000007",role="remote_cluster_client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.245.96",name="instance-0000000020",role="client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.245.96",name="instance-0000000020",role="master"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.245.96",name="instance-0000000020",role="remote_cluster_client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.246.94",name="instance-0000000010",role="client"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.246.94",name="instance-0000000010",role="data_cold"} 1
elasticsearch_nodes_roles{cluster="xxxx",host="172.25.246.94",name="instance-0000000010",role="remote_cluster_client"} 1
```